akka-stream/src/main/java-jdk9-only/akka/stream/javadsl/JavaFlowSupport.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http//www.lightbend.com/>
+ */
+
+package akka.stream.javadsl;
+
+import akka.NotUsed;
+import akka.japi.Pair;
+import akka.japi.Creator;
+import akka.stream.impl.JavaFlowAndRsConverters;
+
+/**
+ * For use only with `JDK 9+`.
+ * <p>
+ * Provides support for `java.util.concurrent.Flow.*` interfaces which mirror the Reactive Streams
+ * interfaces from `org.reactivestreams`. See <a href="http//www.reactive-streams.org/">reactive-streams.org</a>.
+ */
+public final class JavaFlowSupport {
```

Comment:
Do we want to commit on how the final API names will look like already now? I wonder if we could have something more lightweight than `JavaFlowSupport.Source.fromPublisher` but that's mostly a cosmetic question.

Reply:
Ideas? I really liked how this reads so suffered though making it a java class, sadly did not work as nice for java while it was inner `object`s.

I explored quite a bunch of alternatives like: `JavaFlowConverters.sourceFromPublisher` or separate classes `JavaFlowAkkaSourceConverters.fromPublisher` but all ended up being even longer and more difficult to visually distingiush.

The nice thing here is that it reads basically like the current one, and migrating to it is just adding the JavaFlowSupport. The JavaFlow name also does not appear in any actual methods, which i really wanted since the name is kind of crappy... here again I've explored some names but all were I found I found worse: `JdkFlow`, `Java9Flow` (terrible) etc..

Reply:
If you make it a static class, can't you do a static import then?

Reply:
Which "it" specifically? With current names it's not enjoyable; current API would allow importing the specific `fromPublisher` but that reads badly I think: "what from publisher...?"

Importing the classes like so: `import static JavaFlowSupport.*`  in current naming causes conflict hell since that `Source` then is not `javadsl Source`, so that's not a good import to suggest hm.

Did you have other names or imports in mind I'm missing maybe? 🤔